### ISSUE TITLE
Soft or Hard ground-truth masks?

### ISSUE THREAD

**Issue by charleygros** [Project-MONAI/MONAI#721]
Dear MONAI team,

I was wondering: When training or testing a segmentation model: do you binarize the ground-truth masks, that have been transformed by preprocessing / data augmentation operations, just before feeding the network? Is there a parameter that controls this? If so, what are your thoughts on the best practice?

Cheers

**Comment by wyli** [comment]
Hi @charleygros I think if the soft masks carry useful information such as level of confidence then it might be useful to incorporate it in the loss function. In that case there's no need to binarise them, but may require some normalisation to make them to be in a reasonable value range. most of the loss functions in monai don't assume binary labels when "to_one_hot_y" is set to False.

**Comment by charleygros** [comment]
Thanks for your prompt and helpful answer @wyli ! Really appreciate it.

Great to know that loss functions in `monai` do not assume binary labels.

> may require some normalisation to make them to be in a reasonable value range

Do you mean: "make sure that your soft masks values are continuous between 0 and 1"? I can't think of a case where it would go to undesirable values... Am I missing something?

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Could you please help resolve this issue?
Thanks.

**Comment by wyli** [comment]
> Do you mean: "make sure that your soft masks values are continuous between 0 and 1"? I can't think of a case where it would go to undesirable values... Am I missing something?

I think it depends on your problem formulations. could model the output as probabilistic variables in [0, 1] or some confidence scores

**Comment by naga-karthik** [comment]
Hello team, 

Thank you for creating this highly resourceful package. Based on the discussion above, it seems that loss functions can take soft (non-binarized) labels as inputs. However, the DiceMetric as described [here](https://docs.monai.io/en/stable/metrics.html#monai.metrics.DiceMetric), requires `y_pred` to be a binarized tensor, while `y` to be in one-hot format (implying that this also needs to be binarized). I was wondering if there was a specific reason why soft predictions/labels are not used in the computation of the Dice metric (while they are allowed in the calculation of the Dice Loss)? My concern is that in the process of binarizing and converting to one-hot format, we might lose information on the boundaries, which is also essential considering that this DiceMetric is typically used in the validation phase (as seen in the tutorials). 

Is there a way to keep using soft predictions/labels throughout the training/validation phases? Any suggestions on this would be great. Thanks in advance!
